plugins/modules/rds_subnet_group.py

```diff
@@ -128,70 +119,67 @@ def main():
+        module.fail_json_aws(e)
+    # Default.
+    result = create_result(False)
 
     try:
-        exists = False
-        result = create_result(False)
-
-        try:
-            matching_groups = conn.get_all_db_subnet_groups(group_name, max_records=100)
-            exists = len(matching_groups) > 0
-        except BotoServerError as e:
-            if e.error_code != 'DBSubnetGroupNotFoundFault':
-                module.fail_json(msg=e.error_message)
-
-        if state == 'absent':
-            if exists:
-                conn.delete_db_subnet_group(group_name)
-                result = create_result(True)
+        matching_groups = conn.describe_db_subnet_groups(
+            DBSubnetGroupName=group_name, MaxRecords=100).get('DBSubnetGroups')
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        # Boto3 now throws an exception instead of returning an empty list.
+        if e.response['Error']['Code'] == 'DBSubnetGroupNotFoundFault':
+            # No existing subnet, create it if needed, else we can just exit.
+            if state == 'present':
+                try:
+                    new_group = conn.create_db_subnet_group(
+                        DBSubnetGroupName=group_name, DBSubnetGroupDescription=group_description, SubnetIds=group_subnets)
+                    result = create_result(True, new_group)
+                except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+                    module.fail_json_aws(e)
+            module.exit_json(**result)
         else:
-            if not exists:
-                new_group = conn.create_db_subnet_group(group_name, desc=group_description, subnet_ids=group_subnets)
-                result = create_result(True, new_group)
-            else:
-                # Sort the subnet groups before we compare them
-                matching_groups[0].subnet_ids.sort()
-                group_subnets.sort()
-                if (matching_groups[0].name != group_name or
-                        matching_groups[0].description != group_description or
-                        matching_groups[0].subnet_ids != group_subnets):
-                    changed_group = conn.modify_db_subnet_group(group_name, description=group_description, subnet_ids=group_subnets)
-                    result = create_result(True, changed_group)
-                else:
-                    result = create_result(False, matching_groups[0])
-    except BotoServerError as e:
-        module.fail_json(msg=e.error_message)
-
-    module.exit_json(**result)
+            module.fail_json_aws(e)
+    # We have one or more subnets at this point.
+    if state == 'absent':
+        try:
+            conn.delete_db_subnet_group(DBSubnetGroupName=group_name)
+            result = create_result(True)
+            module.exit_json(**result)
+        except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+            module.fail_json_aws(e)
```

Comment:
I know the original code didn't do much with the errors, but it's much better to pass a human readable message along with the exception.  For example:
```suggestion
            module.fail_json_aws(e, msg='Failed to delete a subnet group')
```

Reply:
Fair enough !